### PR TITLE
test(mdc-checkbox): add performance tests for mdc-checkbox

### PR DESCRIPTION
**Pull request by wagnermaciel**
### Context

This branch is checked out of https://github.com/angular/components/pull/20032 because it needs the changes to material-experimental's indigo-pink theme.

### CODE REVIEW

File: test/benchmarks/mdc/checkbox/app.module.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+
+@NgModule({
+  declarations: [CheckboxBenchmarkApp],
+  imports: [
+    BrowserModule,
+    MatCheckboxModule,
+  ],
+  providers: [],
```
Comment (crisbeto) [inline test/benchmarks/mdc/checkbox/app.module.ts:49]: Unnecessary `providers`.

File: test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      id: 'checkbox-click-unchecked-to-checked',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      setup: async () => {
+        await $('#show').click();
+        await $('mat-checkbox').click();
+      },
+      prepare: async () => {
+        await $('mat-checkbox').click();
```
Comment (devversion) [inline test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts:82]: Out of curiosity: Are those `clicks` unnecessary here? When the page loads, the checkbox should be already unchecked, right? Same would apply to other benchmarks here.

File: test/benchmarks/mdc/checkbox/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  selector: 'app-root',
+  template: `
+    <button id="show" (click)="show()">Show</button>
+    <button id="hide" (click)="hide()">Hide</button>
+    <button id="indeterminate" (click)="indeterminate()">Indeterminate</button>
+
+    <mat-checkbox *ngIf="isVisible"
+      [checked]="isChecked"
+      [indeterminate]="isIndeterminate"
+      (change)="toggleIsChecked()">
```
Comment (devversion) [inline test/benchmarks/mdc/checkbox/app.module.ts:25]: Is this `(change)` output needed here? I'd expect the checkbox to toggle without it too.

Comment (wagnermaciel) [reply test/benchmarks/mdc/checkbox/app.module.ts:25]: It is needed to have the checkbox maintain state after being hidden.

File: test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts (hunk 2 above)
Comment (wagnermaciel) [reply test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts:82]: The page is only loaded once for each benchmark. It goes 1. load page 2. call `setup` 3. call `prepare` 4. call `work` 5. repeat 3 and 4 over and over again.

Comment (devversion) [reply test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts:82]: I see, how about having a reset button instead then? It feels not ideal to me that we need to check it initially, and then uncheck it again.

File: test/benchmarks/mdc/checkbox/app.module.ts (hunk 3 above)
Comment (devversion) [reply test/benchmarks/mdc/checkbox/app.module.ts:25]: I see. If we keep it that way we should probably have a comment for it. What do you think?

Also did you explore just leaving it as is. Would it be an issue if hiding/showing resets the state?

Comment (wagnermaciel) [reply test/benchmarks/mdc/checkbox/app.module.ts:25]: Currently I use this to set up the test for render speed of a checked checkbox

Comment (devversion) [reply test/benchmarks/mdc/checkbox/app.module.ts:25]: Okay, that's fair. I'm fine keeping this as is (just a comment would be helpful I assume; but not required).

File: test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts (hunk 2 above)
Comment (wagnermaciel) [reply test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts:82]: I think adding a reset button would be a bit overkill here. I think just clicking once before the prepare/work cycle should be sufficient

Comment (devversion) [reply test/benchmarks/mdc/checkbox/checkbox.perf-spec.ts:82]: Either way works for me, but personally I'd not consider this as readable/clear. It all starts with the ambiguity of `prepare` and `setup`.

**Pull request merged**